### ISSUE TITLE
Shadows overly sharp/pixelated after resolving for some time

### ISSUE THREAD

**Issue by thiagoferronatto** [sultim-t/xash-rt#55]
When constantly moving around, shadows look really nice; sharper when the casting object is close to the surface and softer otherwise. However, after not moving the camera or the player for a few seconds, the image starts to stabilize, a lot of the noise (in darker areas) goes away, and shadows become super sharp and, worse, pixelated. Somehow I don't think this is intended behavior.

Below are some relevant images (more [here](https://imgur.com/a/89nILCD)). These were taken @ 1080p with DLSS v2.5.1 Quality on an RTX 2060 with v528.49 drivers. VSync was off, and so were bloom and lens dirt (although I have no idea if these would have anything to do with the problem).

![Nice soft shadows](https://imgur.com/57pytV3.png)
<sup>Right after moving the player.</sup>

![Pixelated shadows](https://imgur.com/pLZyVXq.png)
<sup>After a few seconds standing still.</sup>

**Comment by sultim-t** [comment]
Yeah, there is a kludge for a direct diffuse lighting variance estimation in the denoiser, because previously there were overly blurred, but now they're overly sharp.
Need to tweak it properly

**Comment by ruvaldak** [comment]
> Yeah, there is a kludge for a direct diffuse lighting variance estimation in the denoiser, because previously there were overly blurred, but now they're overly sharp. Need to tweak it properly

What denoiser do you use? Have you tried Nvidia's NRD? It can separate diffuse and indirect for denoising, using different values for each
